src/main/java/com/aws/greengrass/config/ConfigurationReader.java

```diff
@@ -51,7 +67,9 @@ public static void mergeTLogInto(Configuration config, Reader reader, boolean fo
                         if (mergeCondition != null && !mergeCondition.test(targetTopic)) {
                             continue;
                         }
-                        targetTopic.withNewerValue(tlogline.timestamp, tlogline.value, forceTimestamp);
+                        if (ConfigurationMode.WITH_VALUES.equals(configurationMode)) {
+                            targetTopic.withNewerValue(tlogline.timestamp, tlogline.value, forceTimestamp);
+                        }
```

Comment:
in all config modes it is still modifying the config, removing nodes on L82 for example. Is that desirable?

Reply:
It is intentional, but I don't know if I can change this to make it clearer